gensim/models/phrases.py

```diff
@@ -227,12 +243,17 @@ def export_phrases(self, sentences, out_delimiter=b' ', as_tuples=False):
             then you can debug the threshold with generated tsv
         """
         for sentence in sentences:
-            s = [utils.any2utf8(w) for w in sentence]
             last_bigram = False
             vocab = self.vocab
             threshold = self.threshold
             delimiter = self.delimiter  # delimiter used for lookup
             min_count = self.min_count
+
+            if self.recode_to_utf8:
+                s = [utils.any2utf8(w) for w in sentence]
+            else:
+                s = [utils.any2utf8(w) for w in sentence] if self.is_input_bytes else list(sentence)
```

Comment:
yes, we are not doing `recoding` while training Phrases.

For example,  for `recode_to_utf8=False` and for `unicode input sentences (for training)`, we have `unicode` words in vocab. Now if we provide `bytestrings` tokens in getitem or export_phrases, this mismatch will be a problem [here in phrases.py](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/phrases.py#L289). (or vice-versa)

For, no recoding at all, I think user will have to use same encoded corpus for both training, and phrases retrieval (getitem / export_phrases)